pkg/handlers/namespaces.go

```diff
@@ -19,26 +19,29 @@ import (
 )
 
 // MakeNamespacesLister builds a list of namespaces with an "openfaas" tag, or the default name
-func MakeNamespacesLister(defaultNamespace string, clientset kubernetes.Interface) http.HandlerFunc {
+func MakeNamespacesLister(defaultNamespace string, clusterRole bool, clientset kubernetes.Interface) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		log.Println("Query namespaces")
+		// log.Println("List namespaces")
 
 		if r.Body != nil {
 			defer r.Body.Close()
 		}
 
-		res := ListNamespaces(defaultNamespace, clientset)
+		namespaces := []string{}
+		if clusterRole {
+			namespaces = ListNamespaces(defaultNamespace, clientset)
+		} else {
+			namespaces = append(namespaces, defaultNamespace)
+		}
```

Comment:
We could slightly simplify this using
```suggestion
		namespaces := []string{defaultNamespace}
		if clusterRole {
			namespaces = ListNamespaces(defaultNamespace, clientset)
		}
```

Reply:
Very open to a PR.